### PR TITLE
[2.x] Clone address when setting

### PR DESCRIPTION
**Pull request by Jade-GG**
The issue we ran into was that if you have one address set as both default shipping and billing when entering the checkout, they will both be the same object and thus have the same reference.

If you then choose to change your addresses inside of the checkout and try to change one of the two separately, they would both end up getting changed at the same time (as they are the same object reference), causing you to be unable to actually have a billing address that's different from your shipping address.

This is easily fixed by cloning the address object before setting it here.

I do not believe this is a problem in 3.x as the checkout works entirely different there 🙂

### CODE REVIEW

**Comment by royduin** [comment]
Let's wait for @indykoning his approval

**Review by indykoning — APPROVED** [review]
Looks good, do the addresses still stay in sync when having the "same as shipping" checkbox selected?
(i would assume so since we have a process set to sync the addresses, but just to make sure)

Edit: he's checked the behaviour works as intended ✅

**Pull request merged**
